src/pds/registrysweepers/ancestry/__init__.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from itertools import chain
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from pds.registrysweepers.ancestry.ancestryrecord import AncestryRecord
+from pds.registrysweepers.ancestry.generation import get_bundle_ancestry_records
+from pds.registrysweepers.ancestry.generation import get_collection_ancestry_records
+from pds.registrysweepers.ancestry.generation import get_nonaggregate_ancestry_records
+from pds.registrysweepers.utils import configure_logging
+from pds.registrysweepers.utils import Host
+from pds.registrysweepers.utils import parse_args
+from pds.registrysweepers.utils import write_updated_docs
+
+log = logging.getLogger(__name__)
+
+METADATA_PARENT_BUNDLE_KEY = "ops:Provenance/ops:parent_bundle_identifiers"
+METADATA_PARENT_COLLECTION_KEY = "ops:Provenance/ops:parent_collection_identifiers"
+
+
+def run(
+    base_url: str,
+    username: str,
+    password: str,
+    cross_cluster_remotes=None,
+    verify_host_certs: bool = False,
+    log_filepath: Union[str, None] = None,
+    log_level: int = logging.INFO,
+    registry_mock_query_f: Optional[Callable[[str], Iterable[Dict]]] = None,
+    ancestry_records_accumulator: Optional[List[AncestryRecord]] = None,
+    bulk_updates_sink: Optional[List[Tuple[str, Dict[str, List]]]] = None,
+):
+    configure_logging(filepath=log_filepath, log_level=log_level)
+
+    log.info("Starting ancestry sweeper")
+
+    host = Host(cross_cluster_remotes or [], password, base_url, username, verify_host_certs)
+
+    bundle_records = get_bundle_ancestry_records(host, registry_mock_query_f)
+    collection_records = list(get_collection_ancestry_records(host, registry_mock_query_f))
+    nonaggregate_records = get_nonaggregate_ancestry_records(host, collection_records, registry_mock_query_f)
+
+    ancestry_records = chain(bundle_records, collection_records, nonaggregate_records)
+
+    log.info("Generating document bulk updates for AncestryRecords...")
+    updates: Dict[str, Dict[str, Any]] = {}
```

Comment:
I don't know the expected size of updates document, but at some point we might need to handle them as a buffer of a max size. Let's wait for that to be an issue to implement that. That should not be too complicated anyway to add that feature.

Reply:
@tloubrieu-jpl agreed (on both counts)